test/CMakeLists.txt

```diff
@@ -37,7 +37,7 @@ foreach(prog ${ALL_TESTS})
         target_link_libraries(${prog} PRIVATE SDL2::SDL2main)
     endif()
     target_link_libraries(${prog} PRIVATE SDL2_test)
-    if (BUILD_SHARED_LIBS)
+    if (BUILD_SHARED_LIBS OR NOT TARGET SDL2::SDL2-static)
```

Comment:
(not related to this change)
Perhaps, it makes sense to set a timeout to avoid an accidental stuck test:
https://cmake.org/cmake/help/latest/prop_test/TIMEOUT.html

Reply:
Perhaps, but his branch doesn't run any tests in the CMake build anyway. (#246 does.)

Reply:
#247

Reply:
Thanks!

FYI, the tests are failing on Macos: https://github.com/madebr/SDL_image/runs/6528435637?check_suite_focus=true

Reply:
#248